libraries/chain/db_market.cpp

```diff
@@ -1295,6 +1334,39 @@ asset database::calculate_market_fee( const asset_object& trade_asset, const ass
    return percent_fee;
 }
 
+/**
+ * @brief Calculate the margin fee that is to be taken
+ * @param debt the indebted asset
+ * @param collateral the amount of collateral received (before fees)
+ * @returns the amount of fee that should be collected
+ */
+asset database::calculate_margin_fee(const asset_object& debt, const asset& collateral)const
+{
+   auto ba = debt.bitasset_data(*this);
+   auto price_feed = ba.current_feed;
+   if ( price_feed.settlement_price.is_null() 
+         || price_feed.settlement_price.base.amount == 0
+         || !ba.options.extensions.value.margin_call_fee_ratio.valid())
+      return asset(0);
+   auto ratio = ba.adjusted_mcfr(price_feed);
```

Comment:
It's inefficient to adjust the param on every order. Since it's only related to the price feed, it's best to save an adjusted value whenever the price feed changes or the param itself changes.